nsv13/code/modules/overmap/weapons.dm

```diff
@@ -155,130 +133,90 @@
+		return FALSE
+
+	var/datum/ship_weapon/SW = weapon_types[what]
+	fire_delay = initial(fire_delay) + SW.fire_delay
+	weapon_range = initial(weapon_range) + SW.fire_delay
+	fire_mode = what
 	if(ai_controlled)
 		fire_delay += 10 //Make it fair on the humans who have to actually reload and stuff.
 
-/obj/structure/overmap/proc/fire_railgun(atom/target)
-	if(ai_controlled) //AI ships don't have interiors
-		fire_lateral_projectile(/obj/item/projectile/bullet/railgun_slug, target, 10)
-		return
-	var/proj_type = null //If this is true, we've got a launcher shipside that's been able to fire.
-	var/atom/proj_object = null
-	var/fired = FALSE
-	for(var/X in railguns)
-		if(istype(X, /obj/structure/ship_weapon/railgun))
-			var/obj/structure/ship_weapon/railgun/RG = X
-			if(RG.can_fire())
-				proj_object = RG.fire()
-				if(!proj_object)
-					continue
-				if(istype(proj_object, /obj/item/twohanded/required/railgun_ammo))
-					var/obj/item/twohanded/required/railgun_ammo/RA = proj_object
-					proj_type = RA.proj_type
-					qdel(proj_object)
-				if(proj_type)
-					fired = TRUE
-					break
-	if(!fired)
-		to_chat(gunner, "<span class='warning'>DANGER: Launch failure! Railgun systems are not loaded.</span>")
-		return
-
-
-	var/sound/chosen ='nsv13/sound/effects/ship/railgun_fire.ogg'
-	relay_to_nearby(chosen)
-	flick("railgun_charge",railgun_overlay)
-	shake_everyone(2)
-	if(proj_type)
-		fire_lateral_projectile(proj_type, target, 5)
+	return TRUE
 
-/obj/structure/overmap/proc/shake_everyone(severity)
-	for(var/mob/M in mobs_in_ship)
-		if(M.client)
-			shake_camera(M, severity, 1)
+/obj/structure/overmap/proc/fire_weapon(atom/target, mode=fire_mode, lateral=TRUE) //"Lateral" means that your ship doesnt have to face the target
+	if(ai_controlled || (!linked_areas.len && !main_overmap)) //AI ships and fighters don't have interiors
+		if(fire_mode == FIRE_MODE_TORPEDO) //because fighter torpedoes are special
+			if(fire_torpedo(target))
+				return TRUE
+		else
+			var/datum/ship_weapon/weapon_type = weapon_types[mode]
+			var/obj/proj_type = weapon_type.default_projectile_type
+			for(var/i; i < weapon_type.burst_size; i++)
+				if(lateral)
+					fire_lateral_projectile(proj_type, target)
+				else
+					fire_projectile(proj_type, target)
+				sleep(1)
+			return TRUE
+	else if(weapons[mode] && weapons[mode].len) //It's the main ship, see if any part of our battery can fire
+		for(var/obj/machinery/ship_weapon/SW in weapons[mode])
+			if(SW.can_fire() && SW.fire(target, manual=(mode == fire_mode)))
+				return TRUE
+
+	if(gunner) //Tell them we failed
+		var/datum/ship_weapon/SW = weapon_types[fire_mode]
+		to_chat(gunner, SW.failure_alert)
+	return FALSE
 
 /obj/structure/overmap/proc/fire_torpedo(atom/target)
-	if(!linked_areas.len && !main_overmap) //AI ships don't have interiors
+	if(!linked_areas.len && !main_overmap) //AI ships and fighters don't have interiors
```

Comment:
See my other comment its the same issue. There are probably similar ones elsewhere.

Reply:
Torpedoes don't shake you anyways